imap-proto/src/parser/rfc3501/mod.rs

```diff
@@ -23,6 +23,46 @@ use crate::{
 pub mod body;
 pub mod body_structure;
 
+pub fn parse_response(msg: &[u8]) -> ParseResult {
```

Comment:
I don't want this hoisted up top on its own, we should reorder the entire module at once to flow from high-level to low-level.